elastic/kibana
# [ResponseOps] change event log to use a datastream

## Summary

resolves https://github.com/elastic/kibana/issues/62677

Changes event log from using indices, aliases, and ILM manually, to using [data streams][].

[data streams]: https://www.elastic.co/guide/en/elasticsearch/reference/current/data-streams.html

In general for the event log, we create a new set of indices / aliases / index template for each stack version.  And then when we search over all event logs, we use a wildcard pattern to include all the existing versions.  That continues with this PR.  Just instead of creating indices / aliases for a new version, we create a data stream.

Existing event log indices / aliases are left alone, since the current way to search over existing event logs is with a wildcard.  For instance, `.kibana-event-log-8.7.0` is the alias for 8.7.0 and `.kibana-event-log-8.9.0` will be the data stream for 8.9.0.  Searches over the event log are done using the index pattern `.kibana-event-log-*`, so will search over all the aliases and data streams.

Rolling back versions isn't a problem - the "new" resources created on the migration to a newer version, would only ever be referenced by search, if the deployment was rolled back to a previous verison.  The existing indices / aliases / data streams for that version are left intact, the same structure they were in the release they were created in.

## To test

The basic test is to ensure the event log is stil operational.  You can do by checking the history tab of a running rule, which works off the event log.

To make sure we can still see old events from the older aliases / indices, you should actually start with an older version, create a rule and start it running, note the time the rule was started.  Then migrate that deployment to this PR.  Once Kibana comes up, check that the rule history still includes the run from the previous verison.

### Checklist

Delete any items that are not applicable to this PR.

- [x] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials **(updated README)**
